httpcore/_async/http2.py

```diff
@@ -404,6 +404,14 @@ async def _receive_remote_settings_change(self, event: h2.events.Event) -> None:
     async def _response_closed(self, stream_id: int) -> None:
         await self._max_streams_semaphore.release()
         del self._events[stream_id]
+
+        stream = self._h2_state._get_stream_by_id(stream_id=stream_id)
```

Comment:
Is the usage of private API on the `H2Connection` instance avoidable?

Reply:
yeah, I think the `stream_reset` should only be called in `aclose`(where the request is cancelled by client), thus we no longer need to check the stream closed here. 

Testing code now, an improved version will be pushed later

Reply:
@tomchristie Done. Could you check it again?